syntax_and_semantics/literals/symbol.md

```diff
@@ -1,45 +1,62 @@
 # Symbol
 
-A [Symbol](http://crystal-lang.org/api/Symbol.html) is a constant that is identified by a name without you having to give it a numeric value.
+A [Symbol](http://crystal-lang.org/api/Symbol.html) represents a unique name inside the entire source code.
+It is internally implemented as a constant with a numeric value of type `Int32`.
+
+Symbols are interpreted at compile time and cannot be created dynamically. The only way to create a Symbol is by using a symbol literal, denoted by a colon (`:`) followed by an identifier. The identifier may be enclosed in double quotes (`"`).
 
 ```crystal
-:hello
-:good_bye
+:unquoted_symbol
+:"quoted symbol"
+```
+
+A double-quoted identifier can contain any unicode character including white spaces and accepts the same escape sequences as a [string literal](./string.html), yet no interpolation.
 
-# With spaces and symbols
-:"symbol with spaces"
+An unquoted identifier can contain alphanumeric characters, underscore (`_`) or characters with a code point greater than `159`(`0x9F`). It must not start with a number and may end with an exclamation mark (`!`) or question mark (`?`).
 
-# Ending with question and exclamation marks
+```crystal
 :question?
 :exclamation!
+```
 
-# For the operators
+All [Crystal operators](../operators.html) can be used as symbol names unquoted:
+```crystal
 :+
 :-
 :*
 :/
-:==
-:<
-:<=
-:>
-:>=
-:!
-:!=
-:=~
-:!~
+:%
 :&
 :|
 :^
-:~
 :**
 :>>
 :<<
-:%
+:==
+:!=
+:<
+:<=
+:>
+:>=
+:<=>
+:===
 :[]
 :[]?
 :[]=
-:<=>
-:===
+:!
+:~
+:!~
+:=~
 ```
 
-Internally a symbol is represented as an `Int32`.
+## Percent symbol array literal
+
+Besides the single symbol literal, there is also a percent literal to create an [Array](https://crystal-lang.org/api/Array.html) of symbols. It is indicated by `%i` and a pair of delimiters. Valid delimiters are parenthesis `()`, square brackets `[]`, curly braces `{}`, angles `<>` and pipes `||`. Except for the pipes, all delimiters can be nested meaning a start delimiter inside the string escapes the next end delimiter.
+
+Identifiers may contain any unicode characters. Individual symbols are separated by a single space character (` `) which must be escaped to use it as a part of an identifier.
+
```

Comment:
I would split the escaping example after the initial ones.

And also (before or after the escaping example) I would explain the nested usage with a sample. This way it kind of follows the same pattern as before.